### PR TITLE
View Controller refactor

### PR DESCRIPTION
**Pull request by bdorfman-stripe**
Combines shared VC logic from our screens into base classes: STPCoreViewController, STPCoreScrollViewController, and STPTableViewController. Each inherits from the previous one and does shared logic for that type of VC that was previously copy and pasted around. There's probably even more condensing that can be done of shared logic later.

### CODE REVIEW

**Comment by jack-stripe** [comment]
Small note, I think it's good to keep large style changes like header organization out of changes like this as it adds a bunch of churn to the PR and makes it harder to review.

File: Stripe/STPCoreScrollViewController+Private.h [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ */
+@property (nonatomic, nullable, readonly) UIScrollView *scrollView;
+
+/**
+ This method is used by the base implementation to create the object
+ backing the `scrollView` property. Subclasses can override to change the
+ type of the scroll view (eg UITableView or UICollectionView instead of
+ UIScrollView).
+ */
+- (UIScrollView *)createScrollView;
```
Comment (jack-stripe) [inline Stripe/STPCoreScrollViewController+Private.h]: This can probably be a class method

File: Stripe/STPPaymentMethodsInternalViewController.m [hunk 2]
```diff
@@ -53,20 +52,14 @@ - (instancetype)initWithConfiguration:(STPPaymentConfiguration *)configuration
...
-    [tableView registerClass:[STPPaymentMethodTableViewCell class] forCellReuseIdentifier:STPPaymentMethodCellReuseIdentifier];
-    tableView.sectionHeaderHeight = 30;
-    tableView.separatorStyle = UITableViewCellSeparatorStyleNone; // use fake separator views
-    tableView.separatorInset = UIEdgeInsetsMake(0, 18, 0, 0);
-    self.tableView = tableView;
-    [self.view addSubview:tableView];
+- (void)createAndSetupViews {
+    [super createAndSetupViews];
+
+    self.tableView.allowsMultipleSelectionDuringEditing = NO;
```
Comment (jack-stripe) [inline Stripe/STPPaymentMethodsInternalViewController.m]: I think this and the `separatorInset` call can get moved to STPCoreTableVC?

File: Stripe/STPCoreScrollViewController+Private.h (hunk 1 above)
Comment (bdorfman-stripe) [inline Stripe/STPCoreScrollViewController+Private.h]: Yeah I was leaving it as class for potential future usage that may need it to not be a class method (like init'ing a grouped vs plain style, or init'ing a collection view with a specific layout. Can swap if you want though.

**Comment by bdorfman-stripe** [comment]
Ah yeah sorry, i was going to put the header sorting in a separate pr but i accidentally pushed to this branch and then  ¯\_(ツ)_/¯

File: Stripe/STPPaymentMethodsInternalViewController.m (hunk 2 above)
Comment (bdorfman-stripe) [inline Stripe/STPPaymentMethodsInternalViewController.m]: Nowhere else makes these changes so I wasn't sure? Also `allowsMultipleSelectionDuringEditing` is NO by default anyway.

**Pull request merged**